node-db/lesson1.md

```diff
@@ -1,32 +1,673 @@
-## Week 1: introduction to Node.js
+![](https://img.shields.io/badge/status-draft-darkred.svg) 
+# Node 1
+**What we will learn today?**
+- Node and its ecosystem
+- Express
+- Http
+- JSON format
+- fs
+- This class is based on the [Node Girls Express Workshop](https://github.com/node-girls/express-workshop)
+---
 
-Node.js is a powerful and fast way to write server-side code using JavaScript, our trusty sidekick.
+# What is a server?
 
-### Class
+Servers are computer programs that receive requests from other programs, the *clients* and send back a response e.g share data, information or hardware and software resources.
 
-For our first session, the Node Girls will be running their introductory workshop to Node.js.
+### ...and what is a server in plain English?
 
-The workshop materials are here: https://github.com/node-girls/express-workshop.
+A server is a computer program. Its job is to send and receive data.
 
-Please complete the two tasks in the ["before you start"](https://github.com/node-girls/express-workshop#before-you-start) section before you arrive.
+Let's take a website for example.  A website is just a collection of HTML and CSS files, images, maybe some javascript files. When you type a website address in your browser's address bar, the browser (client) sends a **request** to the server that lives at that address. The browser asks the server to give it the files it needs to display the website properly.
 
-The workshop will start at **10:30 AM** in the usual location. Please make sure to arrive on time!
+![Server flow](https://files.gitter.im/heron2014/FiiK/server.png)
 
-### Homework
+# What is Node.js?
+
+  > [Node.js®](https://nodejs.org/en/) is a **JavaScript runtime** built on Chrome's **V8 JavaScript engine**. Node.js uses an **event-driven**, **non-blocking** I/O model that makes it lightweight and efficient.
```

Comment:
I think it would be awkward to drop those terms without a full explanation of each. If you want to explain what Node is without them you could write something like this...

Your Chrome browser contains a JavaScript runtime (or engine) called V8 and a Web API for performing asynchronous operations such as XHR or `setTimeout`. Node is a copy of that engine but with it's own Web API for performing IO operations.

...it may be worth then introducing what is meant by a single-threaded non-blocking IO as part of a conversation about how a JavaScript runtime is built. It is a really important concept that underpins how the language works.